snippets/base/admin/legacy.py

```diff
@@ -162,6 +162,7 @@ def get_changelist_form(self, request, **kwargs):
         return form
 
     def save_model(self, request, obj, form, change):
+        obj.creator = request.user
```

Comment:
Does this mean that the "creator" will be whoever last saved it? I think you'd need to do:
```suggestion
        if not obj.creator:
            obj.creator = request.user
```

Reply:
of course, great catch!

For `Snippets` we allow `creator` to be null, since we have all these snippets what have been created before we added the field (the new `ASRSnippets` require `creator`). So i changed the code to 

`if not obj.creator and not obj.id`

to find out to my surprise that a new obj already has an `id` by the time it reaches that point, which is before we call `super().save_model()`.

I changed the code to `if not obj.creator and not form.initial` to identify the new objects but I'm curious about your thoughts on this @pmac

Reply:
There is that `change` parameter that is passed to the function that will be `False` if this form is creating a new object and `True` if it's changing an existing one. I'd think you can rely on that, but I'm not sure of the nuance. I suggested doing the more simple thing so that you could give a creator to any snippet that didn't have one already, but if that's not needed  you can probably just change it to:

```python
if not change:
    obj.creator = request.user
```

Reply:
I'll take that, thanks @pmac!

Reply:
I'll take that, thanks @pmac!